indexer/src/main.rs

```diff
@@ -153,34 +157,32 @@ async fn main() -> anyhow::Result<()> {
         std::sync::Arc::new(Mutex::new(SizedCache::with_size(100_000)));
 
     let config: near_lake_framework::LakeConfig = opts.to_lake_config().await;
-    let (sender, stream) = near_lake_framework::streamer(config);
+    let (_, stream) = near_lake_framework::streamer(config);
 
-    tracing::info!(
-        target: INDEXER_FOR_EXPLORER,
-        "Starting Indexer for Explorer (lake)...",
-    );
-    let mut handlers = tokio_stream::wrappers::ReceiverStream::new(stream)
-        .map(|streamer_message| {
-            info!(
-                target: crate::INDEXER_FOR_EXPLORER,
-                "Block height {}", &streamer_message.block.header.height
-            );
-            handle_message(
-                &pool,
-                streamer_message,
-                strict_mode,
-                std::sync::Arc::clone(&receipts_cache),
-            )
-        })
-        .buffer_unordered(1usize);
+    tokio::spawn(async move {
+        tracing::info!(
+            target: INDEXER_FOR_EXPLORER,
+            "Starting Indexer for Explorer (lake)...",
+        );
+        let mut handlers = tokio_stream::wrappers::ReceiverStream::new(stream)
+            .map(|streamer_message| {
+                info!(
+                    target: crate::INDEXER_FOR_EXPLORER,
+                    "Block height {}", &streamer_message.block.header.height
+                );
+                handle_message(
+                    &pool,
+                    streamer_message,
+                    strict_mode,
+                    std::sync::Arc::clone(&receipts_cache),
+                )
+            })
+            .buffer_unordered(1usize);
 
-    while let Some(_handle_message) = handlers.next().await {}
-    drop(handlers); // close the channel so the sender will stop
+        while let Some(_handle_message) = handlers.next().await {}
+    });
 
-    // propagate errors from the sender
-    match sender.await {
-        Ok(Ok(())) => Ok(()),
-        Ok(Err(e)) => Err(e),
-        Err(e) => Err(anyhow::Error::from(e)), // JoinError
-    }
```

Comment:
Had to remove this to ensure that `RecieverStream` would stop when `main` finishes. With this still in place, whenever I exited the process, the indexer would continue to run and I would have to force quit the terminal to stop it.

I'm assuming it had something to do with `sender` not being `drop`ed? This change ensures it is.

Is this correct?

Reply:
I am not exactly sure about correctness. Perhaps @frol can suggest something?